src/librustc/ty/util.rs

```diff
@@ -801,14 +794,14 @@ impl<'a, 'tcx> ty::TyS<'tcx> {
                 // Fixed-length vectors.
                 // FIXME(#11924) Behavior undecided for zero-length vectors.
                 Array(ty, _) => {
-                    is_type_structurally_recursive(tcx, sp, seen, representable_cache, ty)
+                    is_type_structurally_recursive(tcx, seen, representable_cache, ty)
                 }
                 Adt(def, substs) => {
                     // Find non representable fields with their spans
                     fold_repr(def.all_fields().map(|field| {
-                        let ty = field.ty(tcx, substs);
-                        let span = tcx.hir().span_if_local(field.did).unwrap_or(sp);
-                        match is_type_structurally_recursive(tcx, span, seen,
+                        let ty = field.ty(*tcx, substs);
```

Comment:
`FieldDef::ty` invokes a query (`type_of`), if it's not too much fallout, make it take `TyCtxtAt`, too or just leave a comment for the future.

Reply:
There are 17 cases in `librustc` only, where I would have to change `.ty(tcx, _)` -> `ty(tcx.at(DUMMY_SP), _)`.

I would consider this too much fallout. (?)

Reply:
yes, I agree